### PR TITLE
Add support for keys and offset with mouse clicks coming in Capybara 3

### PR DESCRIPTION
**Pull request by twalpole**
This adds support for passing key modifiers and offset x,y coordinates to be set when calling `Node#click`/`right_click`/`double_click` coming in Capybara 3

### CODE REVIEW

File: lib/capybara/webkit/node.rb [hunk 1]
```diff
@@ -68,16 +68,19 @@ def unselect_option
       end
     end
 
-    def click
-      invoke("leftClick")
+    def click(*keys)
+      offset = keys.last.is_a?(Hash) ? keys.pop : {}
```
Comment (jferris) [inline lib/capybara/webkit/node.rb]: Are these arguments coming from Capybara? Would it be possible for Capybara to handle the argument parsing and then provide well-formed arguments to drivers?

File: src/JavascriptInvocation.cpp [hunk 2]
```diff
@@ -41,36 +44,44 @@ InvocationResult JavascriptInvocation::invoke(QWebFrame *frame) {
...
+      leftClick(qm["absoluteX"].toInt(), qm["absoluteY"].toInt(), qm["keys"].value<QVariantList>());
     }
     return InvocationResult(result);
   }
 }
 
-void JavascriptInvocation::leftClick(int x, int y) {
+Qt::KeyboardModifiers JavascriptInvocation::modifiers(const QVariantList& keys){
+  Qt::KeyboardModifiers modifiers = Qt::NoModifier;
+  // for(const QVariant& key: keys){
```
Comment (jferris) [inline src/JavascriptInvocation.cpp]: Looks like this line should be removed?

**Comment by jferris** [comment]
It looks like there weren't any tests added for this change. Since Capybara 3 isn't actually out yet, maybe that makes sense right now, but I assume we want test coverage before merging.

**Comment by twalpole** [comment]
@jferris Test coverage is in the Capybara specs in the master branch - which is currently an allowed failure in the travis matrix.

File: lib/capybara/webkit/node.rb (hunk 1 above)
Comment (twalpole) [reply lib/capybara/webkit/node.rb]: The issue is supporting Capybara < 3 which I assume will still be wanted.  It doesn't pass any options to `click` so the driver needs to accept with or without

Comment (jferris) [reply lib/capybara/webkit/node.rb]: Could it use default arguments? Something like: `click(keys = [], offset = {})`

I guess it's the varargs I'm worried about.

Comment (twalpole) [reply lib/capybara/webkit/node.rb]: Yes it definitely can -- not sure why I was overthinking it -- and that's what review is for -- thanks.

**Pull request merged**